src/fs/directory.rs

```diff
@@ -2,6 +2,14 @@ use crate::runtime::driver::op::Op;
 use std::io;
 use std::path::Path;
 
+/// Creates a directory on the local filesystem.
+/// Returns a future which resolves to unit on success, and ['std::io::Error'] on failure.
+///
+/// Preconditions:
```

Comment:
I'd generally prefer indicating these as modes of failure (see the docs in tokio)